src/test/java/com/synopsys/integration/jenkins/scan/service/DiagnosticsServiceTest.java

```diff
@@ -0,0 +1,26 @@
+package com.synopsys.integration.jenkins.scan.service;
+
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import hudson.FilePath;
+import java.io.File;
+import org.junit.jupiter.api.Test;
+
+public class DiagnosticsServiceTest {
+    private final DiagnosticsService diagnosticsServiceMock = mock(DiagnosticsService.class);
+
+    @Test
+    void archiveArtifactoryTest() {
+        String diagnosticsDir = "/path/to/synopsys-bridge/.bridge";
+        FilePath diagnosticsPath = new FilePath(new File(diagnosticsDir));
+
+        diagnosticsServiceMock.archiveDiagnostics(diagnosticsPath);
+
+        verify(diagnosticsServiceMock, times(1))
+            .archiveDiagnostics(diagnosticsPath);
```

Comment:
1) Refactor this unit test logic! You are supposed to verify whether `artifactArchiver.perform(run, diagnosticsPath, envVars, launcher, listener)` method is called, not the `archiveDiagnostics(FilePath diagnosticsPath)`  method.

2) Add assertion for the `else` condition as well.

Reply:
Refactored as suggested